javatests/arcs/core/analysis/ExpressionClaimDeducerTest.kt

```diff
@@ -0,0 +1,338 @@
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "foo" to Deduction.Path("x"),
+        "bar" to Deduction.Path("y")
+      )
+    )
+  }
+
+  @Test
+  fun new_field_access() {
+    val expr = PaxelParser.parse("new Object {foo: input.foo, bar: input.foo.bar}")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "foo" to Deduction.Path("input", "foo"),
+        "bar" to Deduction.Path("input", "foo", "bar")
+      )
+    )
+  }
+
+  @Test
+  fun sum_variables() {
+    val expr = PaxelParser.parse("x + y")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        listOf("x"),
+        listOf("y")
+      )
+    )
+  }
+
+  @Test
+  fun sum_variables_field_access() {
+    val expr = PaxelParser.parse("x.foo + y.foo.bar")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        listOf("x", "foo"),
+        listOf("y", "foo", "bar")
+      )
+    )
+  }
+
+  @Test
+  fun new_field_access_binexpr() {
+    val expr = PaxelParser.parse("new Object {foo: input.foo, bar: input.foo.bar + input.foo}")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "foo" to Deduction.Path("input", "foo"),
+        "bar" to Deduction.Paths(
+          listOf("input", "foo", "bar"),
+          listOf("input", "foo")
+        )
+      )
+    )
+  }
+
+  @Test
+  fun field_access_nested_expr() {
+    val expr = PaxelParser.parse("(new Foo {x: foo, y: bar}).x")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "x" to Deduction.Path("foo")
+      )
+    )
+  }
+
+  @Test
+  fun field_access_nested_multi_field_expr() {
+    val expr = PaxelParser.parse("(new Foo {a: foo.x + foo.y, b: foo.y + foo.z}).a")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "a" to Deduction.Paths(
+          listOf("foo", "x"),
+          listOf("foo", "y")
+        )
+      )
+    )
+  }
+
+  @Test
+  fun new_nested() {
+    val expr = PaxelParser.parse("new Foo {a: (new Bar {x: cat, y: dog}), b: foo, c: 5}")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "a" to Deduction.Scope(
+          "x" to Deduction.Path("cat"),
+          "y" to Deduction.Path("dog")
+        ),
+        "b" to Deduction.Path("foo"),
+        "c" to Deduction.Path()
+      )
+    )
+  }
+
+  @Test
+  fun new_deeply_nested() {
+    val expr = PaxelParser.parse(
+      "new Foo {x: (new Bar {y: (new Baz {z: (new Buz {a: foo}) }) }) }"
```

Comment:
You could consider multiline for readability:
```
new Foo {
  x: (new Bar {
    y: (new Baz {
      z: (new Buz {
        a: foo})
    })
  })
}
```

Reply:
Looks like `()` are not necessary. If we're using newlines, I'm happy for them to go away.